src/main/groovy/com/mini/asaas/payer/PayerAdapter.groovy

```diff
@@ -0,0 +1,51 @@
+package com.mini.asaas.payer
+
+import com.mini.asaas.enums.PersonType
+import com.mini.asaas.enums.address.AddressState
+import com.mini.asaas.utils.StringUtils
+import com.mini.asaas.utils.Utils
+
+class PayerAdapter {
+
+    String name
+
+    String email
+
+    String cpfCnpj
+
+    String phoneNumber
+
+    PersonType personType
+
+    String address
+
+    String addressNumber
+
+    String complement
+
+    String province
+
+    String city
+
+    AddressState state
+
+    String zipCode
+
+    public PayerAdapter(Map originalParams) {
+        Map<String, String> params = Utils.normalizeParams(originalParams)
+        if (!params) return
+        this.name = params.name
+        this.email = params.email
+        this.cpfCnpj = StringUtils.removeNonNumeric(params.cpfCnpj as String) ?: null
+        this.phoneNumber = StringUtils.removeNonNumeric(params.phoneNumber) ?: null
+        this.personType = PersonType.parseFromCpfCnpj(this.cpfCnpj)
+        this.address = params.address
+        this.addressNumber = StringUtils.removeNonNumeric(params.addressNumber) ?: "S/N"
+        this.complement = params.complement
+        this.province = params.province
+        this.city = params.city
+        this.state = params.state as AddressState
```

Comment:
Melhor assim, que não corre risco de lançar erros de parse. Se o parâmetro não for uma UF válida, será setado como null e aí o erro vai ser capturado na validação do service/constraints e devidamente tratado.

```suggestion
        this.state = AddressState.parseFromUFString(params.state)
```

Reply:
Faz sentido, obrigada